File: src/common/web/integration/authorization/AuthorizationStrategiesCatalog.ts
import { ItemsCatalog } from "../../utils/ItemsCatalog";
import { type Constructable } from "../../utils/Types";
import { AuthorizationStrategy } from "./AuthorizationStrategy";

/**
 * Global catalog of all supported authorization strategies.
 */
@ItemsCatalog.define()
export class AuthorizationStrategiesCatalog extends ItemsCatalog<Constructable<AuthorizationStrategy>> {}

File: src/common/py/integration/authorization/strategies/authorization_strategies_catalog.py
import typing

from .authorization_strategy import AuthorizationStrategy
from ....component import BackendComponent
from ....data.entities.authorization import AuthorizationToken
from ....data.entities.user import UserToken
from ....services import Service
from ....utils import ItemsCatalog

AuthorizationStrategyCreator = typing.Callable[
    [
        BackendComponent,
        Service,
        UserToken | None,
        AuthorizationToken | None,
    ],
    AuthorizationStrategy,
]


@ItemsCatalog.define()
class AuthorizationStrategiesCatalog(ItemsCatalog[AuthorizationStrategyCreator]):
    """
    Global catalog of creator functions for all known authorization strategies.
    """
